File: without_display/others/resize.cpp
#include <opencv2/opencv.hpp>
#include <iostream>

int main(int argc, char* argv[]) {
    if (argc < 2) {
        std::cerr << "Использование: " << argv[0] << " input_video.mp4\n";
        return 1;
    }

    std::string input_path = argv[1];
    std::string output_path = "resized_640x640.mp4";

    // Открытие входного видео
    cv::VideoCapture cap(input_path);
    if (!cap.isOpened()) {
        std::cerr << "Не удалось открыть видеофайл: " << input_path << "\n";
        return 1;
    }

    double fps = cap.get(cv::CAP_PROP_FPS);
    int fourcc = cv::VideoWriter::fourcc('m', 'p', '4', 'v'); // можно заменить на 'XVID', 'H264' и т.п.

    // Создание объекта для записи
    cv::VideoWriter out(output_path, fourcc, fps, cv::Size(640, 640));
    if (!out.isOpened()) {
        std::cerr << "Не удалось открыть выходной файл для записи\n";
        return 1;
    }

    cv::Mat frame, resized;
    while (cap.read(frame)) {
        cv::resize(frame, resized, cv::Size(640, 640));
        out.write(resized);
    }

    cap.release();
    out.release();

    std::cout << "Готово: сохранено в " << output_path << "\n";
    return 0;
}

File: resize.py
import cv2
import sys
import os

# Входной аргумент — путь к видео
if len(sys.argv) != 2:
    print("Usage: python resize_video.py input_video.mp4")
    sys.exit(1)

input_path = sys.argv[1]
output_path = os.path.splitext(input_path)[0] + "_resized_640x640.mp4"

# Целевой размер
WIDTH, HEIGHT = 640, 640

# Открываем входное видео
cap = cv2.VideoCapture(input_path)
if not cap.isOpened():
    print("Error: cannot open video")
    sys.exit(1)

# Получаем FPS и fourcc
fps = cap.get(cv2.CAP_PROP_FPS)
fourcc = cv2.VideoWriter_fourcc(*'mp4v')

# Создаём выходной видеофайл
out = cv2.VideoWriter(output_path, fourcc, fps, (WIDTH, HEIGHT))

frame_count = 0
while True:
    ret, frame = cap.read()
    if not ret:
        break

    resized = cv2.resize(frame, (WIDTH, HEIGHT), interpolation=cv2.INTER_LINEAR)
    out.write(resized)
    frame_count += 1

cap.release()
out.release()

print(f"Готово: сохранено {frame_count} кадров в {output_path}")